src/System.Private.CoreLib/shared/System/Guid.cs

```diff
@@ -415,48 +415,50 @@ private static bool TryParseExactD(ReadOnlySpan<char> guidString, ref GuidResult
                 return false;
             }
 
-            if (guidString[8] != '-' || guidString[13] != '-' || guidString[18] != '-' || guidString[23] != '-')
+            unchecked
             {
-                result.SetFailure(overflow: false, nameof(SR.Format_GuidDashes));
-                return false;
-            }
+                if (guidString[8] != '-' || guidString[13] != '-' || guidString[18] != '-' || guidString[23] != '-')
+                {
+                    result.SetFailure(overflow: false, nameof(SR.Format_GuidDashes));
+                    return false;
+                }
 
-            ref Guid g = ref result._parsedGuid;
+                ref Guid g = ref result._parsedGuid;
+
+                uint uintTmp;
+                uint uintTmp2;
+
+                if (!TryParseHexD(guidString.Slice(0, 4), out uintTmp) ||
+                    !TryParseHexD(guidString.Slice(4, 4), out uintTmp2))
+                    goto FalseExit;
+                g._a = (int)((uintTmp << 16) + uintTmp2);
+
+                if (!TryParseHexD(guidString.Slice(9, 4), out uintTmp) ||
```

Comment:
Using two temporaries in pairs here (and below) probably hurts perf - by using `Unsafe.As` the result can be directly assigned to the correct field.

Reply:
After running benchmarks I am not found noticeable performance changes (at least, current version is not slower than version with Unsafe.As.

Disclaimer: I am have only win-x64 computer to test.